scripts/i18n/update-source.js

```diff
@@ -19,36 +23,42 @@ const dirsToCopy = [
 ]
 
 function cloneOrUpdateRepo(repoName, repoUrl) {
+  shell.cd(path.join(originRepo, `..`))
   if (shell.ls(repoName).code !== 0) {
-    logger.debug(`cloning ${repoName}`)
-    shell.exec(`git clone ${repoUrl}`)
+    logger.debug(`Cloning ${repoName}`)
+    shell.exec(`git clone --quiet --depth 1 ${repoUrl} > /dev/null`)
   } else {
     // if the repo already exists, pull from it
     shell.cd(repoName)
     shell.exec(`git pull origin master`)
-    shell.cd(`..`)
   }
 }
 
-// Copy over contents of origin repo (gatsby) to source repo (gatsby-source-i18n)
-// FIXME make sure the main repo is updated before we do this
+// Copy over contents of origin repo (gatsby) to the source repo (gatsby-source-i18n)
 async function updateSourceRepo() {
   if (shell.cd(cacheDir).code !== 0) {
-    logger.debug(`creating ${cacheDir}`)
+    logger.debug(`Creating ${cacheDir}`)
     shell.mkdir(cacheDir)
     shell.cd(cacheDir)
   }
   cloneOrUpdateRepo(sourceRepo, sourceRepoUrl)
   shell.cd(sourceRepo)
-  // delete old content
+  // Delete old content
   shell.rm(`-rf`, `docs/*`)
-  // repopulate content
+  // Repopulate content
   dirsToCopy.forEach(dir => {
-    shell.cp(`-r`, `../../../../${dir}`, `docs`)
+    shell.cp(`-r`, path.join(originRepo, dir), `docs`)
   })
-  shell.exec(`git add *`)
-  shell.exec(`git commit -m 'Update gatsby monorepo'`)
-  // push to source repo
+
+  // Check if there are any changes to commit
+  if (shell.exec(`git status --porcelain`).stdout.length) {
```

Comment:
I don't think so? We're explicitly only deleting/copying over files in the `docs/` directory only.

Reply:
yes, but lock file will be created by CircleCi when installing deps - and depending where shell will be `cd`ed to - it might try to stage it. I would honestly just commit lock file here - if so probably go with `yarn` and use frozen lock file switch in CircleCI config to avoid having to deal with lock file potentially changing

Reply:
I'm still confused... the script should never be running `yarn` from within `gatsby-i18n-source`, and that's the only repo that's going to be committed against.

Reply:
This probably doesn't apply anymore - I see there is bunch of changes being done here.

But still - installing dependencies here should not cause to produce additional untracked and not ignored files - so we should still add lock files to .gitignore in script or just commit one (preferably yarn.lock) and gitignore other one

Reply:
okay! We already have `yarn.lock` commited. I'll add `package-lock.json` to the .gitignore.

Reply:
My bad, we have lock files in main gitignore, which globally ignores lock files. Sorry for the wrong comments here (and confusion)